src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,7 +1,75 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './FindMovie.scss';
+import classNames from 'classnames';
+import { Movie } from '../../types/Movie';
+import { getMovie } from '../../api';
+import { MovieData } from '../../types/MovieData';
+import { MovieCard } from '../MovieCard';
+
+const movieTemplate = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const FindMovie: React.FC<{
+  setMovies: React.Dispatch<React.SetStateAction<Movie[]>>
+}> = ({ setMovies }) => {
+  const [movie, setMovie] = useState<Movie>(movieTemplate);
+  const [inputValue, setInputValue] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [isNotFind, setIsNotFind] = useState(false);
+
+  const searchFilm = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    e.preventDefault();
+    setIsLoading(false);
+
+    getMovie(inputValue)
+      .then((foundMovie) => {
+        const {
+          Poster = 'https://via.placeholder.com/360x270.png?text=no%20preview',
+          Title,
+          Plot,
+          imdbID,
+        } = foundMovie as MovieData;
+
+        setMovie((prevState) => {
+          return {
+            ...prevState,
+            imgUrl: Poster,
+            title: Title,
+            description: Plot,
+            imdbId: imdbID,
+            imdbUrl: `http://www.imdb/title/${imdbID}`,
+          };
+        });
+
+        if (!Title || Title === movie.title) {
+          setIsNotFind(true);
+        }
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
+
+  const handleFilm = () => {
+    if (movie.title) {
+      setMovies((prevMovie) => {
+        if (prevMovie.some((film) => film.imdbId === movie.imdbId)) {
```

Comment:
If the movie is already present in the list, it would be helpful to notify the user so they know why clicking the button had no effect.